parametric/test_headers_precedence.py

```diff
@@ -0,0 +1,545 @@
+from typing import Any
+
+import pytest
+
+from parametric.spec.tracecontext import get_tracecontext
+from parametric.utils.headers import make_single_request_and_get_inject_headers
+
+parametrize = pytest.mark.parametrize
+
+
+def enable_tracecontext() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE": "tracecontext",
+    }
+    return parametrize("library_env", [env])
+
+
+def enable_datadog() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE": "Datadog",
+    }
+    return parametrize("library_env", [env])
+
+
+def enable_datadog_tracecontext() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE": "Datadog,tracecontext",
+    }
+    return parametrize("library_env", [env])
+
+
+@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("nodejs", "TODO: remove when https://github.com/DataDog/dd-trace-js/pull/2477 lands")
+@pytest.mark.skip_library("python", "BUG: w3c propagation is disabled by default")
```

Comment:
Python should now enable w3c by default. Is this test still failing?

Reply:
I just re-enabled this test case and it's passing! Resolving